logging/jboss/src/test/java/io/quarkus/ts/logging/jboss/DefaultMinLogLevelIT.java

```diff
@@ -25,4 +43,37 @@ public void checkDefaultLogMinLevel() {
         // the value of minimum logging level overrides the logging level
         app.logs().assertDoesNotContain("Trace log example");
     }
+
+    // Verifies https://github.com/quarkusio/quarkus/issues/40016
+    @Test
+    public void checkLogRotationContent() throws IOException {
+        Path logDir = app.getServiceFolder();
+
+        app.given().when().get("/log").then().statusCode(204);
+
+        Awaitility.await().atMost(10, TimeUnit.SECONDS).until(() -> {
+            File[] rotatedFiles = logDir.toFile().listFiles((dir, name) -> name.startsWith(LOG_FILE_NAME));
+            return rotatedFiles != null && rotatedFiles.length > 1;
+        });
+
+        // Verify that the rotated file with the expected suffix exists
+        String expectedSuffix = "." + DATE_FORMATTER.format(LocalDate.now()) + "\\.\\d+$";
+        List<Path> rotatedFiles = Files.list(logDir)
+                .filter(path -> path.getFileName().toString().matches(LOG_FILE_NAME + expectedSuffix)).toList();
+
+        assertFalse(rotatedFiles.isEmpty(), "Rotated log file with expected suffix not found.");
+
+        // Check if any rotated file contains the log message
+        boolean messageFound = rotatedFiles.stream()
+                .flatMap(rotatedFile -> {
+                    try {
+                        return Files.lines(rotatedFile);
+                    } catch (IOException e) {
+                        throw new UncheckedIOException(e);
+                    }
+                })
+                .anyMatch(line -> line.contains(EXPECTED_LOG_MESSAGE));
```

Comment:
this succeeds if at least one of the files has expected line, I don't know how many files you expect. We have `io.quarkus.test.utils.FileUtils`, have you considered using them to cut the verbosity (try catches etc.)? I'd expect using `io.quarkus.test.utils.FileUtils#findTargetFile(java.nio.file.Path, java.lang.String)` and `io.quarkus.test.utils.FileUtils#loadFile(java.lang.String)` would cut your code to three lines.

Reply:
I expect 3 files according to configuration. Check if each of rotated files contain the log message now use `FileUtils.loadFile`

Reply:
okay, 3? in that case previous check was incorrect because you didn't check that **each** file contains that line. You did test that **any** did. I'll comment elsewhere on this expectation.